solutions/devsprint-matheus-reis-2/DeliveryAppChallenge/Service/DeliveryApiError.swift

```diff
@@ -0,0 +1,16 @@
+//
+//  DeliveryApiError.swift
+//  DeliveryAppChallenge
+//
+//  Created by pedro tres on 17/05/22.
+//
+
+import Foundation
+
+enum DeliveryApiError: Error {
+    case invalidURL
+    case errorGeneric(description: String)
+    case invalidResponse
+    case invalidData
+    case errorDecoder
```

Comment:
Essa, além da sugestão linguística, pois `decodingError` descreve melhor a exceção, inclui uma melhoria. Os erros de decodificação, muitas vezes, informam qual atributo do tipo de destino não possui um tipo condizente, por exemplo, então é interessante tirar proveito dessa mensagem de erro.

```suggestion
    case decodingError(description: String)
```